lib/paper_trail/model_config.rb

```diff
@@ -168,6 +168,31 @@ def cannot_record_after_destroy?
         ::ActiveRecord::Base.belongs_to_required_by_default
     end
 
+    def type_aware_has_many(klass)
+      klass.has_many(
+        klass.versions_association_name,
+        lambda do |object|
+          # Support Single Table Inheritance (STI) with custom inheritance columns.
+          #
+          # For example, imagine a `version` whose `item_type` is "Animal". The
+          # `animals` table is an STI table (it has cats and dogs) and it has a
+          # custom inheritance column, `species`. If `attrs["species"]` is "Dog",
+          # type_name is set to `Dog`. If `attrs["species"]` is blank, type_name
+          # is set to `Animal`. You can see this particular example in action in
+          # `spec/models/animal_spec.rb`.
```

Comment:
- Assuming `paper_trail_type_name` can be moved into `RecordTrail`, and renamed to something like `versions_association_item_type`, this would be `object.paper_trail.versions_association_item_type`.
- Extract a local variable `item_type = object.paper_trail.versions_association_item_type` and use in the two places below to avoid calculating it twice.